pricefeed-golang/main.go

```diff
@@ -0,0 +1,122 @@
+package main
+
+import (
+	"context"
+	"log"
+	"math/big"
+	"os"
+	"regexp"
+	"time"
+
+	"github.com/joho/godotenv"
+
+	aggregatorv3 "chainlink-price-feed/aggregatorv3"
+
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/ethclient"
+)
+
+func main() {
+
+	err := godotenv.Load(".env")
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// fetch rpc_url
+	rpcUrl := os.Getenv("RPC_URL")
+	if len(rpcUrl) == 0 {
+		log.Fatal("rpcUrl is empty. check the .env file")
+	}
+
+	// check if chainlink feed proxy is provided in command line
+	args := os.Args[1:]
+	feedAddress := ""
+	if len(args) >= 1 {
+		// 1st argument in command line
+		feedAddress = args[0]
+	} else {
+		// if not provided then check if present in .env file
+		feedAddress = os.Getenv("DEFAULT_FEED_ADDR")
+	}
+
+	client, err := ethclient.Dial(rpcUrl)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// Price Feed address
+	chainlinkPriceFeedProxyAddress, isContract := isContractAddress(feedAddress, client)
```

Comment:
Suggestion:  I think idiomatic Go uses "ok" for booleans that operate as conditional checkers.

See : https://go.dev/doc/effective_go and search for ", ok " on the page.